File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    cnt = {}
    for name in participant:
        if name in cnt:
            cnt[name] += 1
        else:
            cnt[name] = 1
            
    for name in completion:
        if name in cnt:
            cnt[name] -= 1

    for name in cnt:
        if cnt[name] > 0:
            return name

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        Map<String, Integer> map = new HashMap<>();
        
        for(String p: participant) {
            map.put(p, map.getOrDefault(p, 0) + 1);
        }
        
        for(String c: completion) {
            map.put(c, map.get(c) - 1);
        }
        
        for(String n: map.keySet()) {
            if(map.get(n) != 0) {
                return n;
            }
        }
        return null;
    }
}